### PR TITLE
WIP: Add additional detail to appdata files

### PR DESCRIPTION
**Pull request by pwithnall**
In particular, add a `<bundle/>` tag which allows the appdata to be
linked to a flatpak.

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T29392

### CODE REVIEW

**Comment by pwithnall** [comment]
This is WIP and should not be merged, because I’m not sure how best to integrate the substitution of `SDK_BRANCH`, `BRANCH` and `SDK_ARCH` into the build system in this module. Should I rename the `.xml` files to `.xml.in` and add some code to `generate-manifests` which outputs the `.xml` with variables substituted for the current `args.arch`?

File: com.endlessm.BaseEknServicesMultiplexer.metainfo.xml [hunk 1]
```diff
@@ -7,4 +7,6 @@
   <name>Endless Desktop Search Extensions</name>
   <summary>Extension points for Endless Desktop Search</summary>
   <url type="homepage">https://endlessos.com/</url>
+  <project_group>Endless</project_group>
+  <bundle type="flatpak" runtime="com.endlessm.apps.Platform/@SDK_ARCH@/@SDK_BRANCH@" sdk="com.endlessm.apps.Sdk/@SDK_ARCH@/@SDK_BRANCH@">runtime/com.endlessm.EknServicesMultiplexer/@SDK_ARCH@/@BRANCH@</bundle>
```
Comment (dbnicholson) [inline com.endlessm.BaseEknServicesMultiplexer.metainfo.xml:11]: I think this should be BaseEknServicesMultiplexer, right?

**Comment by dnohales** [comment]
I've updated the `generate-manifests` script to do all the replacements and modifications in just one step.

The `eos-build` part of this is here: https://github.com/endlessm/eos-build/pull/1126

Let me know what you think since I'm not really familiar with this project.

File: generate-manifests [hunk 2]
```diff
@@ -3,28 +3,54 @@
 from argparse import ArgumentParser
 import json
 import os
+import re
```
Comment (dbnicholson) [inline generate-manifests:6]: I don't think you ended up using `re` anywhere since you're just doing a `str.replace`.

**Comment by dbnicholson** [comment]
bot, retest this please

File: com.endlessm.BaseEknServicesMultiplexer.metainfo.xml (hunk 1 above)
Comment (pwithnall) [reply com.endlessm.BaseEknServicesMultiplexer.metainfo.xml]: I think Damián fixed this.

File: generate-manifests (hunk 2 above)
Comment (pwithnall) [reply generate-manifests]: Dropped it.

**Comment by pwithnall** [comment]
I addressed the nitpick, fixed a few pyflakes/pylint warnings, and split the one commit in two — one with Damián’s work to refactor `generate-manifests`, and one with my appdata additions.

**Pull request merged**